openwebstart/src/main/java/com/openwebstart/launcher/PhaseTwoWebStartLauncher.java

```diff
@@ -53,6 +56,15 @@ public static void main(String[] args) {
             System.exit(-1);
         }
 
+        if(UpdatePanelConfigConstants.isAutoUpdateActivated(config)) {
+            Executors.newSingleThreadExecutor().execute(() -> {
```

Comment:
Why is this done in a separate thread?
Shouldn't we block the UI until the user has made a decision??

What if he clicks on update when there is already a JNLP application running??

Reply:
As discussed we need to do this in a separate thread. Otherwise each app start would be slower while the updater checks the server for new updates